docs-src/references/dynamic-configuration.md

```diff
@@ -0,0 +1,173 @@
+---
+id: dynamic-configuration
+title: Dynamic configuration reference
+description: Dynamic condifiguration key values can be set to override the default values in a Cluster configuration.
+sidebar_label: Dynamic configuration
+tags:
+  - reference
+---
+
+Temporal Cluster provides [dynamic configuration](/concepts/what-is-cluster-configuration#dynamic-configuration) keys that you can update and apply to a running Cluster without restarting your services.
+
+The dynamic configuration keys are set with default values when you create your Cluster configuration.
+You can override these values as you test your Cluster setup for optimal performance according to your workload requirements.
+
+For the complete list of dynamic configuration keys, see <https://github.com/temporalio/temporal/blob/master/common/dynamicconfig/constants.go>.
+Ensure that you check server releases notes for any changes to these keys and values.
+
+To check the default values set for a dynamic configuration key, check the following links:
+
+- [Frontend Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/frontend/service.go#L176)
+- [History Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/history/configs/config.go#L309)
+- [Matching Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/matching/config.go#L125)
+- [Worker Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/worker/service.go#L193)
+
+Note that setting dynamic configuration is optional.
+Change these values only if you need to override the default values to achieve better performance on your Temporal Cluster.
+Also, ensure that you test your changes before setting these in production.
+
+To override the default dynamic configuration values, specify your custom values and constraints for the dynamic configuration keys that you want to change in a YAML configuration file.
+
+- Each dyanmic configuration key can have zero or more values.
+- Each value can have zero or more constraints.
+- There are only three types of constraints that you can define:
```

Comment:
there are others but probably don't worth mentioning.

Reply:
Updated this section per David's comment.